src/test/java/org/springframework/data/couchbase/domain/AirportJsonValuedObject.java

```diff
@@ -0,0 +1,39 @@
+package org.springframework.data.couchbase.domain;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import org.springframework.data.annotation.PersistenceCreator;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+
+public class AirportJsonValuedObject {
+
+	public AirportJsonValuedObject() {}
+
+	@PersistenceCreator
+	@JsonCreator
+	public AirportJsonValuedObject(String theValue) {
+		this.theValue = reverseMap.get(theValue);
+		if (this.theValue == null) {
+			throw new RuntimeException(
+					"invalid value for " + this.getClass().getName() + " : " + theValue + ". Must be one of " + reverseMap);
+		}
+	}
```

Comment:
This is the equivalent of creating an Enum from the output of its @JsonValue method.